### PR TITLE
Expand the documentation setup

### LINKED ISSUE #34: Create a bibliography page

### PR DESCRIPTION
**Pull request by namurphy**
[PlasmaPy's documentation guide]: https://docs.plasmapy.org/en/stable/development/doc_guide.html

This PR incorporates a bunch of documentation infrastructure from PlasmaPy.  Pretty much all of these points are described in much more detail in [PlasmaPy's documentation guide].  

 - Added `api_static` pages for each module.  These are needed for modules to show up in the documentation (described in [PlasmaPy's documentation guide])
 - Created a file to include reStructuredText links (`docs/_links.rst`)
 - Created a file to include reStructuredText substitutions (`docs/_substitutions.rst`)
 - Updated the index page.
 - Begin a getting started page (`docs/getting_started.rst`)
 - Added an example notebooks page (`docs/examples.rst`) and directory (`docs/notebooks`) with some example galleries.  This is built using nbsphinx.
 - Added a units notebook that was adapted from PlasmaPy
 - Added a sparse bibliography page (with bibtex entries in `bibliography.bib`).  Closes #34.
 - Added an empty glossary page
 - Updated the setup for intersphinx, which allows us to link to objects in other Python documentation pages
 - Added/configured a bunch of Sphinx extensions (see `extensions` in `docs/conf.py`)
 - Added a few docstrings to make the documentation build work (which might lead to a merge conflict with #33)
 - Modified a few imports and some `__all__` lists that were making the documentation build grumpy
 - Updated the documentation tox testing environments (including using the "nitpicky" open which will help us identify broken reStructuredText links before they pile up...not that that happened with any other projects I'm working on 😅). This follows up on #37.
 - Updated some of the requirements, following up on #39.
 - Added some CSS files from PlasmaPy which had been adapted to improve accessibility & color contrast.
 - Changed the theme to `sphinx_rtd_theme`
 
When doing a test build of the docs with `tox -e build_docs`, I started getting an error that a `.genx` file couldn't be found.  My best guess is that this isn't related to the documentation but has more to do with the package itself.

### CODE REVIEW

**Comment by namurphy** [comment]
This is looking just about ready.  I was able to copy/adjust most of the setup from PlasmaPy, so it was pretty straightforward.  

There are still some test failures because `xrtpy` isn't able to find a `.genx` file that it needs.  These test failures were hidden because the full test suite was not being run.  This is causing the documentation builds to fail here, though I was able to get the docs to build locally.  The linters and codespell checks are related to code that's being worked on in #33, so making those changes could end up leading to a messy merge conflict.  In any case, it doesn't look like the test failures are related to this pull request, and we should be able to address them in #33.

File: docs/_links.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+.. _SciPy: https://scipy.org
+.. _SolarSoft IDL: https://www.lmsal.com/solarsoft
+.. _Sphinx: https://www.sphinx-doc.org
+.. _SunPy: https://docs.sunpy.org/en/stable
+.. _tox: https://tox.wiki/en/latest
+.. _X-Ray Telescope: https://xrt.cfa.harvard.edu
+.. _XRT: https://xrt.cfa.harvard.edu/
+.. _XRTpy's documentation: https://xrtpy.readthedocs.io/en/latest
+.. _XRTpy's GitHub repository: https://github.com/HinodeXRT/xrtpy
+.. _Zenodo: https://zenodo.org
```
Comment (namurphy) [inline docs/_links.rst:25]: This file pre-defines a bunch of links that we can use in the online documentation, like...
```
XRTpy is a Python_ package for the `X-Ray Telescope`_ on Hinode_.
```

File: docs/_static/css/admonition_color_contrast.css [hunk 2]
```diff
@@ -0,0 +1,144 @@
+/*
```
Comment (namurphy) [inline docs/_static/css/admonition_color_contrast.css:1]: The CSS files adjust the styles for the online docs, including for font colors.  I copied these two `.css` files over directly from PlasmaPy.  Some of the style settings are for web accessibility (including to make sure that there's enough color contrast between links & the background, etc.).

File: docs/_substitutions.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
+.. |bibliography| replace:: :ref:`bibliography`\
+.. |glossary| replace:: :ref:`glossary`\
+.. |inf| replace:: `~numpy.inf`
+.. |minpython| replace:: 3.8
+.. |nan| replace:: `~numpy.nan`
+.. |ndarray| replace:: :class:`~numpy.ndarray`
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
+.. |TimeDelta| replace:: :class:`~astropy.time.TimeDelta`
+.. |Time| replace:: :class:`~astropy.time.Time`
```
Comment (namurphy) [inline docs/_substitutions.rst:9]: This file includes some reStructuredText substitutions that we can use throughout our docs, like...
```
This function accepts a |Quantity| with units of mass.  If an |ndarray| is provided, SI units will be assumed.
```

File: docs/_substitutions.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+   with the syntax for a substitution followed by an underscore to
+   indicate that it's for a link: |docs/_static|_
+
+.. For these workarounds, if the replacement is something in single back
+   ticks (e.g., `xarray`), then it should also be added to
+   nitpick_ignore_regex in docs/conf.py so that it doesn't get counted
+   as an error in a nitpicky doc build (e.g., tox -e doc_build_nitpicky).
+
+.. _`docs/_static`: https://github.com/PlasmaPy/PlasmaPy/tree/main/docs/_static
+.. |docs/_static| replace:: :file:`docs/_static`
```
Comment (namurphy) [inline docs/_substitutions.rst:22]: These substitutions & links are a workaround for allowing us to format something like a file, but also to get a link to it. There isn't really a natural way to do this in reStructuredText, unfortunately, so this is kind of a mess.

File: docs/api_static/xrtpy.response.channel.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+:orphan:
+
+`xrtpy.response.channel`
+========================
```
Comment (namurphy) [inline docs/api_static/xrtpy.response.channel.rst:4]: The files in `docs/api_static` are to make sure that the functions & classes in these modules show up in the online docs.  It's described more in [PlasmaPy's documentation guide](https://docs.plasmapy.org/en/latest/development/doc_guide.html).

File: docs/conf.py [hunk 6]
```diff
@@ -4,18 +4,16 @@
 # full list see the documentation:
 # http://www.sphinx-doc.org/en/master/config
 
+import os
+
 from datetime import datetime
+from sphinx.application import Sphinx
```
Comment (namurphy) [inline docs/conf.py:10]: `docs/conf.py` is the [configuration file for Sphinx](https://www.sphinx-doc.org/en/master/usage/configuration.html).  If we want to change something about how the documentation is configured, this is usually the file that we need to modify.

File: docs/conf.py [hunk 7]
```diff
@@ -29,53 +27,178 @@
...
+    "astropy": ("https://docs.astropy.org/en/stable/", None),
+    "ndcube": ("https://docs.sunpy.org/projects/ndcube/en/stable/", None),
     "numpy": ("https://numpy.org/doc/stable/", None),
+    "python": ("https://docs.python.org/3", None),
     "scipy": ("https://docs.scipy.org/doc/scipy/reference/", None),
-    "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
-    "astropy": ("https://docs.astropy.org/en/stable/", None),
     "sunpy": ("https://docs.sunpy.org/en/stable/", None),
-    "ndcube": ("https://docs.sunpy.org/projects/ndcube/en/stable/", None),
 }
```
Comment (namurphy) [inline docs/conf.py:81]: Intersphinx is the Sphinx extension that lets us link to the documentation for other projects.  In our docs we can do something like...
```
This function returns a `~numpy.ndarray`.
```

File: docs/examples.rst [hunk 8]
```diff
@@ -0,0 +1,16 @@
+Examples
+========
+
+Here we catalog all the example Jupyter notebooks that have been created for
+the various functionality contained in `xrtpy`.
```
Comment (namurphy) [inline docs/examples.rst:5]: This is the page that we'll eventually use to show a gallery of example Jupyter notebooks.

File: docs/glossary.rst [hunk 9]
```diff
@@ -0,0 +1,10 @@
+.. currentmodule:: xrtpy
+
+.. _glossary:
+
+********
+Glossary
+********
+
+.. glossary::
+   :sorted:
```
Comment (namurphy) [inline docs/glossary.rst:10]: More on the glossary in PlasmaPy's documentation guide...

File: docs/response/index.html [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+.. py:module:: xrtpy.response
+.. currentmodule:: xrtpy.response
+
+Introduction
+============
+
+The `xrtpy.response` subpackage includes functionality for describing
+XRT_ channels & filters, finding the effective area, and calculating the
+temperature response function.
```
Comment (namurphy) [inline docs/response/index.html:15]: This is where the narrative documentation for `xrtpy.response` will need to be added.

File: xrtpy/__init__.py [hunk 11]
```diff
@@ -1,4 +1,7 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+A Python data analysis package for the `X-Ray Telescope`_ (XRT) on
+Hinode_.
+"""
```
Comment (namurphy) [inline xrtpy/__init__.py:4]: I needed to add these module docstrings to get the docs to build okay.

File: docs/glossary.rst (hunk 9 above)
Comment (namurphy) [reply docs/glossary.rst:10]: We can link to a term like: ``` :term:`particle-like` ```

**Review by joyvelasquez — APPROVED** [review]

**Pull request merged**